azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/AvailabilitySets.java

```diff
@@ -3,25 +3,19 @@
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsDeletingByGroup;
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsGettingByGroup;
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsListingByGroup;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsCreating;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsDeleting;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsListing;
+import com.microsoft.azure.management.resources.fluentcore.collection.ResourcesInGroup;
 
 /**
  * Entry point to availability set management API.
  */
-public interface AvailabilitySets extends
-        SupportsListing<AvailabilitySet>,
+public interface AvailabilitySets extends 
+        ResourcesInGroup<AvailabilitySet, AvailabilitySet.DefinitionBlank>,
         SupportsListingByGroup<AvailabilitySet>,
         SupportsGettingByGroup<AvailabilitySet>,
-        SupportsCreating<AvailabilitySet.DefinitionBlank>,
-        SupportsDeleting,
         SupportsDeletingByGroup {
-    /**
-     * Entry point to availability set management API within a specific resource group.
-     */
-    interface InGroup extends
-            SupportsListing<AvailabilitySet>,
-            SupportsCreating<AvailabilitySet.DefinitionCreatable>,
-            SupportsDeleting {}
+    
+    
+    public interface InGroup 
+        extends ResourcesInGroup<AvailabilitySet, AvailabilitySet.DefinitionAfterGroup> {
```

Comment:
it;'s not a question of breaking change from the user's pov. It's for us to avoid issues like: originally the interface says DefinitionCreatable is the next thing after .withGroup. But then for one reason or another, we change it in the interface so that .withGroup is followed by a new DefinitionWithFoo, not DefinitionCreatable. Your impl then still points at DefinitionCreatable. So you have to fix both the impl and the interface. With DefinitionAfterGroup, you only need to update the interface file. Less change and more contained. Easier to deal with. 
Also, the impl should not care about what the required vs optional params are, and it should not care about the order of the params. That's the interface's job. The DefinitionAfterGroup thing allows us to enforce that separation.